language/stdlib/modules/LibraAccount.move

```diff
@@ -1161,20 +1225,29 @@ module LibraAccount {
             TransactionFee::pay_fee(Libra::withdraw(coin, transaction_fee_amount))
         }
     }
+
+    /// Epilogue for WriteSet trasnaction
+    fun writeset_epilogue(
+        lr_account: &signer,
+        writeset_payload: vector<u8>,
+        writeset_sequence_number: u64
+    ) acquires LibraWriteSetManager, LibraAccount, Balance {
+        let t_ref = borrow_global_mut<LibraWriteSetManager>(CoreAddresses::LIBRA_ROOT_ADDRESS());
+
+        Event::emit_event<Self::UpgradeEvent>(
+            &mut t_ref.upgrade_events,
+            UpgradeEvent { writeset_payload },
+        );
+        // Currency code don't matter here as it won't be charged anyway.
+        epilogue<LBR::LBR>(lr_account, writeset_sequence_number, 0, 0, 0);
```

Comment:
The `0` uglyness still resides. Do we want to make clean separation of incrementing sequence number only? IMHO having multiple function will introduce extra borrow_global. Do we want to trade that for the code cleaness?

Reply:
I think it's fine like this. Keep it simple....